public/app/plugins/datasource/dashboard/datasource.ts

```diff
@@ -20,8 +25,65 @@ export class DashboardDatasource extends DataSourceApi<DashboardQuery> {
     return `Dashboard Reference: ${query.panelId}`;
   }
 
-  query(options: DataQueryRequest<DashboardQuery>): Promise<DataQueryResponse> {
-    return Promise.reject('This should not be called directly');
+  query(options: DataQueryRequest<DashboardQuery>): Observable<DataQueryResponse> {
+    const scene: SceneObject | undefined = options.scopedVars?.__sceneObject?.value;
+
+    if (!scene) {
+      throw new Error('Can only be called from a scene');
+    }
+
+    const query = options.targets[0];
+    if (!query) {
+      return of({ data: [] });
+    }
+
+    const panelId = query.panelId;
+
+    if (!panelId) {
+      return of({ data: [] });
+    }
+
+    let sourcePanel = this.findSourcePanel(scene, panelId);
+
+    if (!sourcePanel) {
+      return of({ data: [], error: { message: 'Could not find source panel' } });
+    }
+
+    let sourceDataProvider = sourcePanel?.state.$data;
+
+    if (query.withTransforms && sourceDataProvider instanceof SceneDataTransformer) {
+      sourceDataProvider = sourceDataProvider.state.$data!;
+    }
+
+    if (!sourceDataProvider || !sourceDataProvider.getResultsStream) {
+      return of({ data: [] });
+    }
+
+    if (!sourceDataProvider?.isActive) {
+      sourceDataProvider?.activate();
+      sourceDataProvider.setContainerWidth!(500);
+    }
```

Comment:
this needs to be backed into the Observable so we can activate on subscriber start and unactivate when the stream is completed/closed. But we can do that in a follow-up PR